### ISSUE TITLE
Guidance on Troubleshooting

### ISSUE THREAD

**Issue by delmendo** [commanded/commanded#381]
I'm currently running Commanded in a K8s cluster at Digital Ocean.  I'm troubleshooting a problem where I trigger and event , but the projector never kicks in.  I can query the database from the code, but the event never gets picked up by a projector until I redeploy.  I shrunk the cluster down to a single node and set everything to local -pubsub/registry.  

I have consistency: set to  :eventual everywhere - both the dispatch and the projections.

The code works fine in development. I've watched the logs from the elixir and I never get the projection until I send out a new image and it triggers the event stream again.  Is there a way to manually trigger projectors? Is there a log setting that might show me what is preventing them from processing? I turned the main app's log up to debug and it is just silent .  Any guidance would be super appreciated.

**Comment by slashdotdash** [comment]
The Elixir EventStore uses Postgres’ LISTEN / NOTIFY feature to notify subscribers whenever new events are persisted. It’s likely that this is not working if you see event handlers processing existing events on restart (as the subscription will initially read events on startup before switching to receive the via the pub/sub approach). 

Are you hosting Postgres yourself or using a managed Postgres service provided by Digital Ocean? You could try running the EventStore test suite against the database to determine whether it is supported.

**Comment by slashdotdash** [comment]
If you discover that LISTEN / NOTIFY is not working on your Postgres database let me know and I will take a look at a polling option. It’s a feature that has been on the backlog for a while and should solve the problem.

**Comment by falti** [comment]
Curious if LISTEN / NOTIFY isn't active by default postgres installations? I was considering Digital Ocean as a low cost alternative and wonder if this could be a show stopper.
@slashdotdash thanks for this awesome software!

**Comment by delmendo** [comment]
Ok you have unlocked the piece I was missing. I'm using the managed db with of pg bouncer pooling on. The recommend config is transaction pooling. https://www.pgbouncer.org/features.html that specifically breaks notify listen. Do connections to to repo and the event store need listen or just one of them? I wasn't familiar with that feature of postgres so I didn't realize it would be effected by the pooling. I'll test turing session pooling on to see if it works with pooling at all

**Comment by delmendo** [comment]
So if you do the non listen notify - would it just switch to a pubsub via redis ( since everything else seems to use that )

**Comment by slashdotdash** [comment]
Currently only LISTEN / NOTIFY is supported  for subscriptions. Adding polling as a fallback option for environments which don't support Postgres notifications will require code changes to EventStore. I will add an issue to track this new feature.

**Comment by slashdotdash** [comment]
I’ve created issue https://github.com/commanded/eventstore/issues/205

**Comment by delmendo** [comment]
OK I spent the day mucking with my configuration.  I'm using the managed database postgresql at Digital Ocean. @falti  Listen/Notify is totally supported with a direct connection to the database. DO offers a connection pooling system (PGBouncer). The recommended configuration for that is transactional.  I was able to setup the EventStore to talk directly to the managed db and pointed the read store to the connection pool.  This seems to work correctly -  my projections are firing off and things seem to be working properly. Now I can focus on scaling this back up to a bunch of nodes. Thanks @slashdotdash  for such prompt guidance!
